### PR TITLE
Laravel 10 and updates

### PR DESCRIPTION
**Pull request by sabas**
- Upgrade dependencies to Laravel 10
- Migrate SMTP Adapter to use Symfony Mail directly
- [BREAKING] Use Blade templates in content: the {{content}} placeholder now needs to be written as {!! $content !!}

### CODE REVIEW

**Comment by ppolacek** [comment]
Thanks for your PR. An error (undefined variable) occurs when using templates (on the campaign preview page) - specifically if I want to use more placeholders besides the basic "content" (for example "email" or "first_name").

That's because you're passing only the "content" variable when using Blade::render() method - see 
https://github.com/mettle/sendportal-core/pull/187/files#diff-6c7b787be1493ddfe8a8f8063122e80e5a251e3087949348725e89d43f6f77a2R262

This should not occur later when sending the messages, as all tags are generated then.
https://github.com/mettle/sendportal-core/pull/187/files#diff-1b913a0ad17255f035ef3f36d75495934814ffe2eabec21ff21c90b683696813R47

Thanks

File: src/Models/Campaign.php [hunk 1]
```diff
@@ -255,7 +256,12 @@ public function getBounceRatioAttribute()
     public function getMergedContentAttribute(): ?string
     {
         if ($this->template_id) {
-            return str_replace(['{{content}}', '{{ content }}'], $this->content, $this->template->content);
+            return Blade::render(
+                $this->template->content,
+                [
+                    'content' => $this->content
```
Comment (ppolacek) [inline src/Models/Campaign.php:262]: https://github.com/mettle/sendportal-core/pull/187#issuecomment-1590633349

```
return Blade::render(
                $this->template->content,
                [
                    'content' => $this->content,
                    // placeholders
                    'email' => '{!! $email !!}',
                    'first_name' => '{!! $first_name !!}',
                    'last_name' => '{!! $last_name !!}',
                    'unsubscribe_url' => '{!! $unsubscribe_url !!}',
                    'webview_url' => '{!! $webview_url !!}',
                ]
            );
```

**Comment by sabas** [comment]
@ppolacek if you want please add a commit to my PR :) (I am not actively using it right now)

File: src/Models/Campaign.php (hunk 1 above)
Comment (sabas) [reply src/Models/Campaign.php]: Thanks. I made a commit!

**Comment by damms005** [comment]
Is there anything blocking this PR that I can contribute to?

**Comment by shankhadevpadam** [comment]
Why this repo is inactive?

File: src/Models/Campaign.php (hunk 1 above)
Comment (shankhadevpadam) [reply src/Models/Campaign.php]: I got this error.

```
Undefined constant "email"
```

**Comment by shankhadevpadam** [comment]
@sabas 

I got this error.

```
Undefined constant "email"
```

**Comment by sabas** [comment]
@shankhadevpadam what line? I don't actively use it so feel free to improve the PR.

**Comment by shankhadevpadam** [comment]
> @shankhadevpadam what line? I don't actively use it so feel free to improve the PR.

Models\Campaign.php

```
/**
 * Get the merged content for this email, including the template content.
 */
public function getMergedContentAttribute(): ?string
{
    if ($this->template_id) {
        return Blade::render(
            $this->template->content,
            [
                'content' => $this->content,
                // placeholders
                'email' => '{!! $email !!}',
                'first_name' => '{!! $first_name !!}',
                'last_name' => '{!! $last_name !!}',
                'unsubscribe_url' => '{!! $unsubscribe_url !!}',
                'webview_url' => '{!! $webview_url !!}',
            ]
        );
    }

    return $this->content;
}
```

**Comment by ppolacek** [comment]
Hi @sabas, I know you don't actively use this fork, this is mainly for informing others.

After a few months of working with this fork (unfortunately, I've modified/added a lot of things already, so I can't edit this PR directly), I've discovered several bugs - using the `Blade::render` method in `MergeContentService` (and elsewhere) is not ideal for 2 reasons:
- it breaks compatibility, the [original placeholders](https://sendportal.io/docs/v2/features/templates) are used without the $ sign, if using Blade they must be used as `$variables`
- if using templates, if `$content` contains HTML code (which it 99% does), the template needs to display `$content` as unescaped `{!! $content !!}` - you mentioned this in your first post - except that after this step, all other variables in `$content` will be displayed in raw format (`$email` will be displayed as `$email`, not `email@example.com`). That's why I would recommend going back to the original solution using `str_ireplace()` fn - then everything will work as it should, and compatibility will be preserved.

There are a few more details, e.g. here the [$templateContent](https://github.com/mettle/sendportal-core/pull/187/files#diff-1b913a0ad17255f035ef3f36d75495934814ffe2eabec21ff21c90b683696813R97) parameter should also be nullable `?string`.

But otherwise this fork helped me quite a lot, so thank you 🍺

**Comment by sabas** [comment]
@ppolacek congratulations! do you plan to release a proper fork? 🙂

**Comment by ppolacek** [comment]
Thanks! I don't know yet, I added a lot of things like full SMTP tracking, web IMAP client (email communication with subscribers directly via SP), possibility to have multiple email contents/variants within one campaign with advanced timing, possibility to send emails from multiple accounts within a campaign, custom unsubscribe logic, etc. etc.
Anyway, there is still a lot of work to be done and quite a few things are not 100% complete, so, not yet

**Pull request closed without merging**